src/firmware.c

```diff
@@ -25,11 +27,34 @@
 
 uint32_t __stack_chk_guard = 0;
 
+/**
+ * Sets the correct memory configuration for this firmware.
+ *
+ * Extends the APPDATA_1 MPU region up to the end of the
+ * flash memory, and sets it to R/W.
+ *
+ * FUTURE: maybe add SmartEEPROM support here?
+ */
+static void _memory_init(void)
+{
+    uint32_t rbar;
+    uint32_t rasr;
+    rbar = (FLASH_APPDATA_START + (FLASH_APPDATA_LEN / 2)) | MPU_REGION_VALID |
+           MPU_REGION_NUMBER_APPDATA_1;
+    rasr = MPU_REGION_ENABLE | MPU_REGION_TYPE_NORMAL | MPU_REGION_EXECUTE_NEVER |
+           mpu_region_size(FLASH_APPDATA_LEN) | MPU_REGION_READ_WRITE;
+    /* Remove the current region setting. */
+    mpu_disable_region(MPU_REGION_NUMBER_APPDATA_1);
+    /* Reset the last selected region to the new setting. */
+    mpu_set_region(rbar, rasr);
+}
+
 int main(void)
 {
     init_mcu();
     system_init();
     peripherals_init();
+    _memory_init();
```

Comment:
This should be in `common_main()` I think, as it should be handled the same way in the factorysetup firmware.

Reply:
Fixed.